tests/ms_azure/test_utils.py

```diff
@@ -228,10 +230,114 @@ def test_update_new_skus_gen1_default(self, technical_config_obj: VMIPlanTechCon
         )
         assert res == expected
 
-    def test_update_existing_skus_gen2_default(
+    def test_update_new_skus_arm64(
+        self,
+        disk_version_arm64_obj: DiskVersion,
+    ) -> None:
+        """Ensure the creation of 1 new SKU for ARM64 from scratch."""
+        expected = [
+            VMISku.from_json(x)
+            for x in [
+                {"imageType": "arm64Gen2", "skuId": "plan1-arm64", "security_type": None},
+            ]
+        ]
+        res = update_skus(
+            disk_versions=[disk_version_arm64_obj],
+            generation="V2",
+            plan_name="plan1",
+        )
+        assert res == expected
+
+    def test_update_new_skus_mixed_x64_arm64_gen2_default(
+        self,
+        disk_version_arm64_obj: DiskVersion,
+        technical_config_obj: VMIPlanTechConfig,
+    ) -> None:
+        """Ensure the creation of 3 new SKUs: 2 for x64 (V2/V1) and 1 for ARM64 from scratch."""
+        expected = [
+            VMISku.from_json(x)
+            for x in [
+                {"imageType": "x64Gen2", "skuId": "plan1", "security_type": None},
+                {"imageType": "arm64Gen2", "skuId": "plan1-arm64", "security_type": None},
+                {"imageType": "x64Gen1", "skuId": "plan1-gen1", "security_type": None},
+            ]
+        ]
+        res = update_skus(
+            disk_versions=[disk_version_arm64_obj] + technical_config_obj.disk_versions,
+            generation="V2",
+            plan_name="plan1",
+        )
+        assert res == expected
+
+    def test_update_new_skus_mixed_x64_arm64_gen1_default(
+        self,
+        disk_version_arm64_obj: DiskVersion,
+        technical_config_obj: VMIPlanTechConfig,
+    ) -> None:
+        """Ensure the creation of 3 new SKUs: 2 for x64 (V1/V2) and 1 for ARM64 from scratch."""
+        expected = [
+            VMISku.from_json(x)
+            for x in [
+                {"imageType": "x64Gen1", "skuId": "plan1", "security_type": None},
+                {"imageType": "arm64Gen2", "skuId": "plan1-arm64-gen2", "security_type": None},
```

Comment:
@lslebodn this is a tiny issue on the naming convention in the way I've implemented the changes...

Whenever the default generation comes as `V1` the ARM64 version will be named `PLAN-arm64-gen2`, unfortunately, instead of `PLAN-arm64`... I think that's something we may not worry but it's outside the common naming convention...

I believe it will increase the code complexity to fix that, besides, if the x84 Gen1 is the `default` for that offer, any other `Gen2` would be alternate, as the ARM64 in this case, so.... It's kinda correct (expected) I guess. 

Just FYI